spec/models/post_spec.rb

```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  before(:each) do
+    User.create(name: 'User1', email: 'user1@example.com',
+                password: '123456', password_confirmation: '123456')
+  end
+
+  describe 'Post validations' do
+    it 'post cannot be empty' do
+      user = User.find_by(name: 'User1')
+      post = user.posts.build
+      expect(post.valid?).to be(false)
+    end
+  end
+
+  describe 'Post associations' do
+    it 'deletes dependent objects' do
+      user = User.find_by(name: 'User1')
+      expect(user.class).to be(User)
+      post = user.posts.create(content: 'A simple post')
+      expect(post.likes.count).to eq(0)
```

Comment:
- Kindly make sure to have one expect per it block as it helps you be specific about what you are testing. [optional]